File: 01 Introdução/1040 Média 3/1040.py
notas = input().split(' ')
A, B, C, D = [float(nota) for nota in notas]

media = (A*2+B*3+C*4+D)/10
print(f'Media: {media:.1f}')
if media >= 7.0:
    resultado = 'Aluno aprovado.'
elif 7 > media >= 5:
    resultado = 'Aluno em exame.'
else:
    resultado = 'Aluno reprovado.'

print(resultado)
if resultado == 'Aluno em exame.':
    nota_do_exame = float(input())
    print(f"Nota do exame: {nota_do_exame:.1f}")
    media = (media + nota_do_exame) / 2
    if media >= 5:
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")
    print(f"Media final: {media}")

# https://www.beecrowd.com.br/judge/pt/problems/view/1040
# Ranking 02143º, Time 0.000, Submiss Date 12/02/2023 23:15:07 (Python 3.9)

File: 01 Introdução/1040 Média 3/1040.java
import java.util.Scanner;

class Main {

  public static void main(String args[]) {
    Scanner sc = new Scanner(System.in);

    float A = sc.nextFloat();
    float B = sc.nextFloat();
    float C = sc.nextFloat();
    float D = sc.nextFloat();

    float media = (A * 2 + B * 3 + C * 4 + D) / 10;
    String resultado;

    System.out.printf("Media: %.1f\n", media);

    if (media >= 7) {
      resultado = "Aluno aprovado.";
    } else if (7 > media && media >= 5) {
      resultado = "Aluno em exame.";
    } else {
      resultado = "Aluno reprovado.";
    }

    System.out.println(resultado);

    if (resultado == "Aluno em exame.") {
      float notaDoExame = sc.nextFloat();
      System.out.printf("Nota do exame: %.1f\n", notaDoExame);
      media = (media + notaDoExame) / 2;
      if (media >= 5) {
        System.out.println("Aluno aprovado.");
      } else {
        System.out.println("Aluno reprovado.");
      }
      System.out.printf("Media final: %.1f\n", media);
    }

    sc.close();
  }
}